new/property.md

```diff
@@ -0,0 +1,223 @@
+- `MyResource.state_attrs` will include the property name.
+- `MyResource.properties` will contain the property type.
+- The setter and getter manipulate the class variable `@<name>`.
+
+(This is the same as before, except the addition of the `properties` hash.)
+
+#### `attribute`
+
+`attribute` will remain on LWRPs, and be an alias to `property` with no distinctions.
+
+#### Use `property` instead of `attribute` in documentation
+
+`attribute` will continue to be supported; there are simply too many things in the world to deprecate it. However, any generic documentation that talks about attributes will be renamed to talk about properties. `attribute` itself will still be documented.
+
+We will need to write a comprehensive resource writing guide as well, in order to get it pumped to the top of Google, so that `attribute` comes up less and less often in searches and `property` comes up more and more.
+
+### Property type
+
+Properties with a single type are common enough that we support a "type" for a property, specifiable after its name.
+
+```ruby
+class MyResource < Chef::Resource
+  property :content, String
+end
+```
+
+This is actually an alias for `is` (described later here).
+
+#### Reusing types
+
+The type of a property is represented by `Chef::Resource::PropertyType`, and contains accessors for all the properties of a type (`must_be`, `name_attribute`, `kind_of`, etc.).
+
+When you declare `property :name, <type>, <options>`, one of two things happens:
+
+- If the type is a PropertyType instance, it is dup'd and any <option>s are set on the new type.
+- If the type is a Class, a new PropertyType instance is created with `kind_of` set to [Class] and any options are set on the new type.
+- If type is not passed, a new PropertyType instance is created, and any options are set on the new type.
+
+#### Property Inheritance
+
+Subclasses get properties from their parent.
+
+```ruby
+class A < Chef::Resource
+  property :a, String
+end
+class B < Chef::Resource
+  property :b, String
+end
+B.state_attrs -> [ :a, :b ]
+```
+
+#### Overriding Properties
+
+When a property is overridden, the override is *complete*: that is, the parent
+type is not extended or mixed in any way.
+
+```ruby
+class A < Chef::Resource
+  property :a, String, default: 'Hello'
+end
+class B < A
+  property :a
+end
+A.properties[:a].default #=> 'Hello'
+B.properties[:a].default #=> nil
+```
+
+#### `is`
+
+`is` is a new validation parameter that uses Ruby's match operator `===` (the thing that drives `case` and `when`).
+
+```ruby
+# These are equivalent
+property :x, [ :a, :b, :c ]
+property :x, is: [ :a, :b, :c ]
+```
+
+It is worth noting that many existing validations can be expressed directly in terms of `is`:
+
+```ruby
+# These:
+attribute :path, kind_of: String
+attribute :path, equal_to: [ :a, :b, :c ]
+attribute :path, regex: /^\//
+attribute :path, respond_to: :merge
+attribute :path, cannot_be: :empty
+property :path, is: String
```

Comment:
I think you meant to have something like `# Are equivalent to these:`, overall this example is hard to read though.